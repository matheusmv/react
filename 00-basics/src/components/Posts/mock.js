export const postsMock = {
  posts: [
    { id: 1, title: 'title 1', body: 'body 1', cover: 'img/img-1.png' },
    { id: 2, title: 'title 2', body: 'body 2', cover: 'img/img-2.png' },
    { id: 3, title: 'title 3', body: 'body 3', cover: 'img/img-3.png' },
    { id: 4, title: 'title 4', body: 'body 4', cover: 'img/img-4.png' },
  ],
};
