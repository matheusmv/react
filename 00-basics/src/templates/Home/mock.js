export const mockPosts = [
  {
    id: 1,
    title: 'title 1',
    body: 'body 1',
  },
  {
    id: 2,
    title: 'title 2',
    body: 'body 2',
  },
  {
    id: 3,
    title: 'title 3',
    body: 'body 3',
  },
  {
    id: 4,
    title: 'title 4',
    body: 'body 4',
  },
  {
    id: 5,
    title: 'title 5',
    body: 'body 5',
  },
  {
    id: 6,
    title: 'title 6',
    body: 'body 6',
  },
  {
    id: 7,
    title: 'title 7',
    body: 'body 7',
  },
  {
    id: 8,
    title: 'title 8',
    body: 'body 8',
  },
  {
    id: 9,
    title: 'title 9',
    body: 'body 9',
  },
  {
    id: 10,
    title: 'title 10',
    body: 'body 10',
  },
  {
    id: 11,
    title: 'title 11',
    body: 'body 11',
  },
  {
    id: 12,
    title: 'title 12',
    body: 'body 12',
  },
  {
    id: 13,
    title: 'title 13',
    body: 'body 13',
  },
];

export const mockPhotos = [
  {
    url: 'img/img-1.jpg',
  },
  {
    url: 'img/img-2.jpg',
  },
  {
    url: 'img/img-3.jpg',
  },
  {
    url: 'img/img-4.jpg',
  },
  {
    url: 'img/img-5.jpg',
  },
  {
    url: 'img/img-6.jpg',
  },
  {
    url: 'img/img-7.jpg',
  },
  {
    url: 'img/img-8.jpg',
  },
  {
    url: 'img/img-9.jpg',
  },
  {
    url: 'img/img-10.jpg',
  },
  {
    url: 'img/img-11.jpg',
  },
  {
    url: 'img/img-12.jpg',
  },
  {
    url: 'img/img-13.jpg',
  },
];
