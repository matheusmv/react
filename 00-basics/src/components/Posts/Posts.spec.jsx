import React from 'react';

import { render, screen } from '@testing-library/react';
import { Posts } from '.';
import { postsMock as props } from './mock';

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', /title/i)).toHaveLength(4);
    expect(screen.getAllByRole('img', /title/i)).toHaveLength(4);
    expect(screen.getAllByText(/body/i)).toHaveLength(4);

    expect(screen.getByRole('img', { name: /title 1/i })).toHaveAttribute('src', 'img/img-1.png');
  });

  it('should not render posts', () => {
    render(<Posts />);

    expect(screen.queryByRole('heading', /title/i)).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);

    expect(container).toMatchSnapshot();
  });
});
