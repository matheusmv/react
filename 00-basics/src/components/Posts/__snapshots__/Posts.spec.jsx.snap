// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="title 1"
        src="img/img-1.png"
      />
      <div
        class="post-content"
      >
        <h3>
          1
           
          title 1
        </h3>
        <p>
          body 1
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title 2"
        src="img/img-2.png"
      />
      <div
        class="post-content"
      >
        <h3>
          2
           
          title 2
        </h3>
        <p>
          body 2
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title 3"
        src="img/img-3.png"
      />
      <div
        class="post-content"
      >
        <h3>
          3
           
          title 3
        </h3>
        <p>
          body 3
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title 4"
        src="img/img-4.png"
      />
      <div
        class="post-content"
      >
        <h3>
          4
           
          title 4
        </h3>
        <p>
          body 4
        </p>
      </div>
    </div>
  </div>
</div>
`;
